source/tutorials/nixos-configuration-on-vm.md

```diff
@@ -0,0 +1,169 @@
+
+The `hardware-configuration.nix` file is specific to the hardware the `nix-generate-config` is being run on.
+We can ignore that file for this tutorial because it has no effect inside a virtual machine.
+
+The `configuration.nix` file contains various suggestions for the initial setup of a desktop computer.
+Without comments the configuration file contains the following content:
+```nix
+{ config, pkgs, ... }:
+{
+  imports =
+    [
+      ./hardware-configuration.nix
+    ];
+
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  system.stateVersion = "22.05";
+}
+```
+
+In this tutorial we focus on testing NixOS configurations on a virtual machine, in these cases you can remove the `hardware-configuration.nix` import.
+
+Changes to the configuration need to be positioned inside the curly bracket.[^bracket]
+
+To be able to log in, you must uncomment the section that specifies the user `alice`, or add the following lines:
+```nix
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ];
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+  };
+```
+
+Additionally, you need to specify a password for this user.
+An insecure[^password] solution is to specify an plain text password by adding the `initialPassword` option to the user configuration:
+```nix
+    initialPassword = "testpw";
+```
+
+The complete `configuration.nix` file now looks like this:
+```nix
+{ config, pkgs, ... }:
+{
+  boot.loader.systemd-boot.enable = true;
+  boot.loader.efi.canTouchEfiVariables = true;
+
+  services.xserver.enable = true;
+
+  services.xserver.displayManager.gdm.enable = true;
+  services.xserver.desktopManager.gnome.enable = true;
+
+  users.users.alice = {
+    isNormalUser = true;
+    extraGroups = [ "wheel" ]; # Enable ‘sudo’ for the user.
+    packages = with pkgs; [
+      firefox
+      thunderbird
+    ];
+    initialPassword = "testpw";
+  };
+
+  system.stateVersion = "22.05";
+}
+```
+
+## Creating a QEMU based virtual machine using a configuration
+
+A virtual machine is created with the `nix-build` command.
+
+To select `configuration.nix` in the working directory, specify the configuration file as an argument:
+```shell-session
+$ nix-build '<nixpkgs/nixos>' -A vm -I nixos-config=./configuration.nix
+```
+[^nixosrebuild]
+
+This command builds the attribute `vm` utilizing the version of Nixpkgs as specified in the environment variable `NIX_PATH` and using the NixOS configuration as specified in the relative path.
+
+## Running the virtual machine
+
+The previous command creates a link with the name `result` in the working directory.
+It links to the folder that contains the virtual machine.
+
+```shell-session
+$ ls -R ./result
```

Comment:
Do not prepend shell prompts, otherwise it's either not clear whether (or incurs additional cognitive overhead) to exclude it when copy-pasting.

```suggestion
ls -R ./result
```